lib/streamDAO.js

```diff
@@ -53,6 +53,25 @@ function filterDatumForMongo(datum) {
       datum.deviceId = datum.deviceId.replace(BadDeviceIDPrefix, GoodDeviceIDPrefix);
       datum.id = schema.generateId(datum, schema.idFields(datum.type));
     }
+
+    // This is a migration step which converts times to native time objects before write to keep client changes minimal
+    // it would be better if this could be done right at input, but its simpler this way.
+    entry = datum;
+    while (entry) {
+	  if ('time' in entry && typeof(entry.time) === 'string') {
+	    entry.time = new Date(entry.time);
+	  }
+
+	  if ('createdTime' in entry && typeof(entry.createdTime) === 'string') {
+	    entry.createdTime = new Date(entry.createdTime);
+	  }
+
+	  if ('archivedTime' in entry && typeof(entry.archivedTime) === 'string') {
+		entry.archivedTime = new Date(entry.archivedTime);
+	  }
+
+	  entry = entry.suppressed;
```

Comment:
Should you convert dates in `datum.previous` the same you convert them in `datum.suspended`?